### PR TITLE
Add support for API 1.9

### PR DESCRIPTION
**Pull request by rubickcz**

### CODE REVIEW

File: pycsob/client.py [hunk 1]
```diff
@@ -102,9 +102,6 @@ def payment_init(self, order_no, total_amount, return_url, description, merchant
         :return: response from gateway as OrderedDict
         """
 
-        if len(description) > 255:
```
Comment (rubickcz) [inline pycsob/client.py:105]: The parameter `description` is still used to create shopping cart contents, but is no longer sent directly, so this check is not needed anymore.

File: pycsob/client.py [hunk 2]
```diff
@@ -102,14 +102,14 @@ def payment_init(self, order_no, total_amount, return_url, description, merchant
...
-            raise ValueError('Description length is over 255 chars')
+        if len(description) > 20:
```
Comment (rubickcz) [inline pycsob/client.py:105]: Parameter `description` is now used only to create "cart" data, so I changed the validation accordingly.

File: pycsob/client.py [hunk 3]
```diff
@@ -127,7 +127,6 @@ def payment_init(self, order_no, total_amount, return_url, description, merchant
             ('returnUrl', return_url),
             ('returnMethod', return_method),
             ('cart', cart),
-            ('description', description),
```
Comment (rubickcz) [inline pycsob/client.py:130]: Removed in API 1.9

File: pycsob/client.py [hunk 4]
```diff
@@ -219,7 +218,8 @@ def customer_info(self, customer_id):
         r = self._client.get(url)
         return utils.validate_response(r, self.pubkey)
 
-    def oneclick_init(self, orig_pay_id, order_no, total_amount, currency='CZK', description=None):
+    def oneclick_init(self, orig_pay_id, order_no, total_amount, currency='CZK', description=None,
+                      return_url='http://localhost', return_method='GET', client_initiated=False):
```
Comment (rubickcz) [inline pycsob/client.py:222]: Parameters `return_url` and `return_method` are mandatory, but in fact are not needed, when `client_initiated=False`. See https://github.com/csob/platebnibrana/issues/26.

File: pycsob/client.py [hunk 5]
```diff
@@ -233,8 +233,11 @@ def oneclick_init(self, orig_pay_id, order_no, total_amount, currency='CZK', des
             ('totalAmount', total_amount),
             ('currency', currency),
             ('description', description),
+            ('returnUrl', return_url),
+            ('returnMethod', return_method),
+            ('clientInitiated', client_initiated),
```
Comment (rubickcz) [inline pycsob/client.py:238]: Fun fact: Order of these fields **does** matter. This order is used to create signature. Wrong order will result in wrong signature.

File: pycsob/client.py [hunk 6]
```diff
@@ -233,8 +233,11 @@ def oneclick_init(self, orig_pay_id, order_no, total_amount, currency='CZK', des
...
         ))
-        url = utils.mk_url(base_url=self.base_url, endpoint_url='payment/oneclick/init')
+        url = utils.mk_url(base_url=self.base_url, endpoint_url='oneclick/init')
```
Comment (rubickcz) [inline pycsob/client.py:240]: Some URLs changed in API 1.9

File: pycsob/utils.py [hunk 7]
```diff
@@ -2,7 +2,7 @@
 import re
 from base64 import b64encode, b64decode
 from collections import OrderedDict
-from Crypto.Hash import SHA
+from Crypto.Hash import SHA256
```
Comment (rubickcz) [inline pycsob/utils.py:5]: New sign algorithm in API 1.9

**Review by Formulka — APPROVED** [review]

**Pull request merged**